### ISSUE TITLE
can't serialize object so can't see exception

### ISSUE THREAD

**Issue by thedavidmeister** [boot-clj/boot-cljs#177]
```
Compiling ClojureScript...
• index.html.js
                                       java.lang.Thread.run                  Thread.java:  745
         java.util.concurrent.ThreadPoolExecutor$Worker.run      ThreadPoolExecutor.java:  617
          java.util.concurrent.ThreadPoolExecutor.runWorker      ThreadPoolExecutor.java: 1142
                        java.util.concurrent.FutureTask.run              FutureTask.java:  266
                                                        ...                                   
                        clojure.core/binding-conveyor-fn/fn                     core.clj: 2027
                              adzerk.boot-cljs/compile-1/fn                boot_cljs.clj:  160
                                   adzerk.boot-cljs/compile                boot_cljs.clj:   72
                                          boot.pod/call-in*                      pod.clj:  413
                                                        ...                                   
org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.invoke  ClojureRuntimeShimImpl.java:  102
org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl.invoke  ClojureRuntimeShimImpl.java:  109
                                                        ...                                   
                                          boot.pod/call-in*                      pod.clj:  410
                                      boot.pod/eval-fn-call                      pod.clj:  359
                                         clojure.core/apply                     core.clj:  657
                                                        ...                                   
                         adzerk.boot-cljs.impl/compile-cljs                     impl.clj:  158
                     adzerk.boot-cljs.util/serialize-object                     util.clj:   56
                     java.io.ObjectOutputStream.writeObject      ObjectOutputStream.java:  348
                    java.io.ObjectOutputStream.writeObject0      ObjectOutputStream.java: 1178
             java.io.ObjectOutputStream.writeOrdinaryObject      ObjectOutputStream.java: 1432
                 java.io.ObjectOutputStream.writeSerialData      ObjectOutputStream.java: 1496
                java.io.ObjectStreamClass.invokeWriteObject       ObjectStreamClass.java: 1028
                                                        ...                                   
                            java.lang.Throwable.writeObject               Throwable.java:  985
              java.io.ObjectOutputStream.defaultWriteObject      ObjectOutputStream.java:  441
              java.io.ObjectOutputStream.defaultWriteFields      ObjectOutputStream.java: 1548
                    java.io.ObjectOutputStream.writeObject0      ObjectOutputStream.java: 1178
             java.io.ObjectOutputStream.writeOrdinaryObject      ObjectOutputStream.java: 1432
                 java.io.ObjectOutputStream.writeSerialData      ObjectOutputStream.java: 1496
                java.io.ObjectStreamClass.invokeWriteObject       ObjectStreamClass.java: 1028
                                                        ...                                   
                            java.lang.Throwable.writeObject               Throwable.java:  985
              java.io.ObjectOutputStream.defaultWriteObject      ObjectOutputStream.java:  441
              java.io.ObjectOutputStream.defaultWriteFields      ObjectOutputStream.java: 1548
                    java.io.ObjectOutputStream.writeObject0      ObjectOutputStream.java: 1178
             java.io.ObjectOutputStream.writeOrdinaryObject      ObjectOutputStream.java: 1432
                 java.io.ObjectOutputStream.writeSerialData      ObjectOutputStream.java: 1496
                java.io.ObjectStreamClass.invokeWriteObject       ObjectStreamClass.java: 1028
                                                        ...                                   
                            java.lang.Throwable.writeObject               Throwable.java:  985
              java.io.ObjectOutputStream.defaultWriteObject      ObjectOutputStream.java:  441
              java.io.ObjectOutputStream.defaultWriteFields      ObjectOutputStream.java: 1548
                    java.io.ObjectOutputStream.writeObject0      ObjectOutputStream.java: 1178
             java.io.ObjectOutputStream.writeOrdinaryObject      ObjectOutputStream.java: 1432
                 java.io.ObjectOutputStream.writeSerialData      ObjectOutputStream.java: 1509
              java.io.ObjectOutputStream.defaultWriteFields      ObjectOutputStream.java: 1548
                    java.io.ObjectOutputStream.writeObject0      ObjectOutputStream.java: 1178
             java.io.ObjectOutputStream.writeOrdinaryObject      ObjectOutputStream.java: 1432
                 java.io.ObjectOutputStream.writeSerialData      ObjectOutputStream.java: 1509
              java.io.ObjectOutputStream.defaultWriteFields      ObjectOutputStream.java: 1548
                    java.io.ObjectOutputStream.writeObject0      ObjectOutputStream.java: 1174
                      java.io.ObjectOutputStream.writeArray      ObjectOutputStream.java: 1378
                    java.io.ObjectOutputStream.writeObject0      ObjectOutputStream.java: 1184
java.io.NotSerializableException: clojure.spec.alpha$regex_spec_impl$reify__1200
```

**Comment by thedavidmeister** [comment]
```
                         adzerk.boot-cljs.impl/compile-cljs                     impl.clj:  158
                     adzerk.boot-cljs.util/serialize-object                     util.clj:   56
```
these two lines point to this i think https://github.com/boot-clj/boot-cljs/blob/master/src/adzerk/boot_cljs/impl.clj#L161

**Comment by thedavidmeister** [comment]
I checked out a local copy of boot-cljs and stuck a `prn` directly in `impl.clj` and managed to extract the exception that way

**Comment by Deraen** [comment]
I don't have any idea on how to improve exception serialization.

**Comment by thedavidmeister** [comment]
@Deraen it's throwing an exception while trying to do the serialization, but a simple prn worked

**Comment by thedavidmeister** [comment]
e.g.

```
      (catch Exception e
       (let [ex (handle-ex e source-paths dirs)
             serialized-ex (try (util/serialize-object ex)
                            (catch Exception e
                             (throw (Exception. (pr-str ex)))))]
        {:exception (util/serialize-object ex)})))))
```

**Comment by flyboarder** [comment]
I am also running into this issue, need someway to serialize Clojure Spec errors
```
java.lang.Exception: #error {
                      :cause "Call to hoplon.core/defattr did not conform to spec:\nIn: [1] val: [elem key value] fails spec: :hoplon.spec/attr-args at: [:args :args :elem] predicate: simple-symbol?\n"
                      :data {:clojure.spec.alpha/problems [{:path [:args :args :elem], :pred clojure.core/simple-symbol?, :val [elem key value], :via [:hoplon.spec/defattr :hoplon.spec/attr-args], :in [1]}], :clojure.spec.alpha/spec #object[clojure.spec.alpha$regex_spec_impl$reify__2436 0xac3fb58 "clojure.spec.alpha$regex_spec_impl$reify__2436@ac3fb58"], :clojure.spec.alpha/value (:baz [elem key value] (elem :attr {key value})), :clojure.spec.alpha/args (:baz [elem key value] (elem :attr {key value}))}
                      :via
                      [{:type clojure.lang.ExceptionInfo
                        :message "tst/src/cljs/hoplon/defattr_test.cljs [line 7, col 1] Call to hoplon.core/defattr did not conform to spec:\nIn: [1] val: [elem key value] fails spec: :hoplon.spec/attr-args at: [:args :args :elem] predicate: simple-symbol?\n"
                        :data {:clojure.spec.alpha/args (:baz [elem key value] (elem :attr {key value})), :boot.util/omit-stacktrace? true, :file "tst/src/cljs/hoplon/defattr_test.cljs", :column 1, :line 7, :clojure.spec.alpha/value (:baz [elem key value] (elem :attr {key value})), :from :boot-cljs, :tag :cljs/analysis-error, :clojure.spec.alpha/problems [{:path [:args :args :elem], :pred clojure.core/simple-symbol?, :val [elem key value], :via [:hoplon.spec/defattr :hoplon.spec/attr-args], :in [1]}], :clojure.spec.alpha/spec #object[clojure.spec.alpha$regex_spec_impl$reify__2436 0xac3fb58 "clojure.spec.alpha$regex_spec_impl$reify__2436@ac3fb58"]}
                        :at [clojure.core$ex_info invokeStatic "core.clj" 4739]}
                       {:type clojure.lang.ExceptionInfo
                        :message "failed compiling file:/Users/matt/.boot/cache/tmp/Users/matt/Projects/hoplon/hoplon/1nll/-ru88jt/hoplon/defattr_test.cljs"
                        :data {:file #object[java.io.File 0x2bed2a62 "/Users/matt/.boot/cache/tmp/Users/matt/Projects/hoplon/hoplon/1nll/-ru88jt/hoplon/defattr_test.cljs"]}
                        :at [clojure.core$ex_info invokeStatic "core.clj" 4739]}
                       {:type clojure.lang.ExceptionInfo
                        :message "Call to hoplon.core/defattr did not conform to spec:\nIn: [1] val: [elem key value] fails spec: :hoplon.spec/attr-args at: [:args :args :elem] predicate: simple-symbol?\n at line 7 /Users/matt/.boot/cache/tmp/Users/matt/Projects/hoplon/hoplon/1nll/-ru88jt/hoplon/defattr_test.cljs"
                        :data {:file "/Users/matt/.boot/cache/tmp/Users/matt/Projects/hoplon/hoplon/1nll/-ru88jt/hoplon/defattr_test.cljs", :line 7, :column 1, :tag :cljs/analysis-error}
                        :at [clojure.core$ex_info invokeStatic "core.clj" 4739]}
                       {:type clojure.lang.ExceptionInfo
                        :message "Call to hoplon.core/defattr did not conform to spec:\nIn: [1] val: [elem key value] fails spec: :hoplon.spec/attr-args at: [:args :args :elem] predicate: simple-symbol?\n"
                        :data {:clojure.spec.alpha/problems [{:path [:args :args :elem], :pred clojure.core/simple-symbol?, :val [elem key value], :via [:hoplon.spec/defattr :hoplon.spec/attr-args], :in [1]}], :clojure.spec.alpha/spec #object[clojure.spec.alpha$regex_spec_impl$reify__2436 0xac3fb58 "clojure.spec.alpha$regex_spec_impl$reify__2436@ac3fb58"], :clojure.spec.alpha/value (:baz [elem key value] (elem :attr {key value})), :clojure.spec.alpha/args (:baz [elem key value] (elem :attr {key value}))}
                        :at [clojure.core$ex_info invokeStatic "core.clj" 4739]}]
                      :trace
                      [[clojure.core$ex_info invokeStatic "core.clj" 4739]
                       [clojure.core$ex_info invoke "core.clj" 4739]
                       [clojure.spec.alpha$macroexpand_check invokeStatic "alpha.clj" 689]
                       [clojure.spec.alpha$macroexpand_check invoke "alpha.clj" 681]
                       [clojure.lang.Var invoke "Var.java" 385]
                       [cljs.analyzer$macroexpand_1_STAR_ invokeStatic "analyzer.cljc" 3284]
                       [cljs.analyzer$macroexpand_1_STAR_ invoke "analyzer.cljc" 3271]
                       [cljs.analyzer$macroexpand_1 invokeStatic "analyzer.cljc" 3333]
                       [cljs.analyzer$macroexpand_1 invoke "analyzer.cljc" 3329]
                       [cljs.analyzer$analyze_seq invokeStatic "analyzer.cljc" 3366]
                       [cljs.analyzer$analyze_seq invoke "analyzer.cljc" 3346]
                       [cljs.analyzer$analyze_form invokeStatic "analyzer.cljc" 3536]
                       [cljs.analyzer$analyze_form invoke "analyzer.cljc" 3532]
                       [cljs.analyzer$analyze_STAR_ invokeStatic "analyzer.cljc" 3585]
                       [cljs.analyzer$analyze_STAR_ invoke "analyzer.cljc" 3576]
                       [cljs.analyzer$analyze invokeStatic "analyzer.cljc" 3610]
                       [cljs.analyzer$analyze invoke "analyzer.cljc" 3593]
                       [cljs.compiler$emit_source invokeStatic "compiler.cljc" 1377]
                       [cljs.compiler$emit_source invoke "compiler.cljc" 1356]
                       [cljs.compiler$compile_file_STAR_$fn__4637 invoke "compiler.cljc" 1456]
                       [cljs.compiler$with_core_cljs invokeStatic "compiler.cljc" 1271]
                       [cljs.compiler$with_core_cljs invoke "compiler.cljc" 1260]
                       [cljs.compiler$compile_file_STAR_ invokeStatic "compiler.cljc" 1441]
                       [cljs.compiler$compile_file_STAR_ invoke "compiler.cljc" 1434]
                       [cljs.compiler$compile_file$fn__4668 invoke "compiler.cljc" 1538]
                       [cljs.compiler$compile_file invokeStatic "compiler.cljc" 1513]
                       [cljs.compiler$compile_file invoke "compiler.cljc" 1489]
                       [cljs.closure$compile_file invokeStatic "closure.clj" 540]
                       [cljs.closure$compile_file invoke "closure.clj" 531]
                       [cljs.closure$eval6923$fn__6924 invoke "closure.clj" 609]
                       [cljs.closure$eval6859$fn__6860$G__6848__6867 invoke "closure.clj" 493]
                       [cljs.closure$eval6929$fn__6930 invoke "closure.clj" 618]
                       [cljs.closure$eval6859$fn__6860$G__6848__6867 invoke "closure.clj" 493]
                       [cljs.closure$compile_sources$iter__7047__7051$fn__7052 invoke "closure.clj" 954]
                       [clojure.lang.LazySeq sval "LazySeq.java" 40]
                       [clojure.lang.LazySeq seq "LazySeq.java" 49]
                       [clojure.lang.Cons next "Cons.java" 39]
                       [clojure.lang.RT next "RT.java" 706]
                       [clojure.core$next__5108 invokeStatic "core.clj" 64]
                       [clojure.core$dorun invokeStatic "core.clj" 3134]
                       [clojure.core$doall invokeStatic "core.clj" 3140]
                       [clojure.core$doall invoke "core.clj" 3140]
                       [cljs.closure$compile_sources invokeStatic "closure.clj" 950]
                       [cljs.closure$compile_sources invoke "closure.clj" 939]
                       [cljs.closure$build invokeStatic "closure.clj" 2596]
                       [cljs.closure$build invoke "closure.clj" 2507]
                       [cljs.build.api$build invokeStatic "api.clj" 205]
                       [cljs.build.api$build invoke "api.clj" 189]
                       [adzerk.boot_cljs.impl$compile_cljs invokeStatic "impl.clj" 154]
                       [adzerk.boot_cljs.impl$compile_cljs invoke "impl.clj" 113]
                       [clojure.lang.AFn applyToHelper "AFn.java" 156]
                       [clojure.lang.AFn applyTo "AFn.java" 144]
                       [clojure.lang.Var applyTo "Var.java" 702]
                       [clojure.core$apply invokeStatic "core.clj" 657]
                       [clojure.core$apply invoke "core.clj" 652]
                       [boot.pod$eval_fn_call invoke "pod.clj" 359]
                       [boot.pod$call_in_STAR_ invoke "pod.clj" 410]
                       [clojure.lang.Var invoke "Var.java" 381]
                       [org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl invoke "ClojureRuntimeShimImpl.java" 109]
                       [org.projectodd.shimdandy.impl.ClojureRuntimeShimImpl invoke "ClojureRuntimeShimImpl.java" 102]
                       [sun.reflect.NativeMethodAccessorImpl invoke0 "NativeMethodAccessorImpl.java" -2]
                       [sun.reflect.NativeMethodAccessorImpl invoke "NativeMethodAccessorImpl.java" 62]
                       [sun.reflect.DelegatingMethodAccessorImpl invoke "DelegatingMethodAccessorImpl.java" 43]
                       [java.lang.reflect.Method invoke "Method.java" 498]
                       [clojure.lang.Reflector invokeMatchingMethod "Reflector.java" 93]
                       [clojure.lang.Reflector invokeInstanceMethod "Reflector.java" 28]
                       [boot.pod$call_in_STAR_ invoke "pod.clj" 413]
                       [adzerk.boot_cljs$compile invokeStatic "boot_cljs.clj" 72]
                       [adzerk.boot_cljs$compile invoke "boot_cljs.clj" 60]
                       [adzerk.boot_cljs$compile_1$fn__4300 invoke "boot_cljs.clj" 160]
                       [clojure.core$binding_conveyor_fn$fn__5476 invoke "core.clj" 2022]
                       [clojure.lang.AFn call "AFn.java" 18]
                       [java.util.concurrent.FutureTask run "FutureTask.java" 266]
                       [java.util.concurrent.ThreadPoolExecutor runWorker "ThreadPoolExecutor.java" 1142]
                       [java.util.concurrent.ThreadPoolExecutor$Worker run "ThreadPoolExecutor.java" 617]
                       [java.lang.Thread run "Thread.java" 745]]}
```

**Comment by Deraen** [comment]
Should be fixed in 2.1.5
